src/pages/experiments/[experimentId]/plots-and-tables/frequency/index.jsx

```diff
@@ -101,7 +105,23 @@ const FrequencyPlotPage = ({ experimentId }) => {
   if (!config) {
     return <Skeleton />;
   }
+  const formatCSVData = (plotData) => {
+    const newCsvData = [];
+
+    louvainClusters[0].children.forEach((cluster) => {
+      const entriesForCluster = plotData.filter((entry) => entry.yCellSetKey === cluster.key);
+      const cellSetName = cellSets.properties[cluster.key].name;
+      const newEntry = { 'Louvain Cluster': cellSetName };
+      entriesForCluster.forEach((entry) => {
+        const sampleName = cellSets.properties[entry.x].name;
+        newEntry[sampleName] = entry.y;
+      });
+      newCsvData.push(newEntry);
+    });
 
+    setCsvFilename(`${experimentName.replace(/ /g, '_')}-freq-plot-${config.frequencyType}`);
```

Comment:
Can you refactor this functionality for creating the filename into a new function? we'll definitely use it in the future.

Reply:
You mean in  a new file? Or function in the file? How will we be reusing this same filename?

Reply:
Or you mean having a file with all the fileNames generators - seems a bit overkill to me

Reply:
There are other tickets in the future that will implement CSV download (e.g. https://biomage.atlassian.net/browse/BIOMAGE-1571), it would be good to reuse this functionality.